pkg/array/array.go

```diff
@@ -109,7 +119,7 @@ func (s *Locker) UpdateArrays(configPath string, fs fs.Interface) error {
 		return fmt.Errorf("can't get config for arrays: %s", err.Error())
 	}
 	s.SetArrays(arrays)
-	IPToArray = matcher
+	updateIPToArray(matcher)
```

Comment:
You are not really updating, you are setting so should start the function with set*.